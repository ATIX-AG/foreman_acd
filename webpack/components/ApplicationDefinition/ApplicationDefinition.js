import $ from 'jquery';
import React, { useState } from 'react'
import PropTypes from 'prop-types';
import {
  Icon,
  Button,
} from 'patternfly-react';
import * as resolve from 'table-resolver';
import ForemanModal from 'foremanReact/components/ForemanModal';
import Select from 'foremanReact/components/common/forms/Select';
import ParameterSelection from '../ParameterSelection';
import AddTableEntry from '../common/AddTableEntry';
import DeleteTableEntry from '../common/DeleteTableEntry';
import RailsData from '../common/RailsData';
import EasyHeaderFormatter from '../common/EasyHeaderFormatter';
import AnsiblePlaybookSelector from './components/AnsiblePlaybookSelector';

import {
  Table,
  FormControl,
  inlineEditFormatterFactory,
} from 'patternfly-react';

import {
  PARAMETER_SELECTION_PARAM_TYPE_FOREMAN,
  PARAMETER_SELECTION_PARAM_TYPE_ANSIBLE,
} from '../ParameterSelection/ParameterSelectionConstants';

class ApplicationDefinition extends React.Component {

  constructor(props) {
    super(props);
  }

  isEditing({rowData}) {
    return (rowData.backup !== undefined);
  }

  createAnsibleGroupObject(ansibleGroups, withAll=false) {
    const ansibleGroupObj = {};

    const ansibleGroupArray = Object.keys(ansibleGroups);
    ansibleGroupArray.forEach(e => (ansibleGroupObj[e] = e));

    if ((withAll === false) && (ansibleGroupObj.hasOwnProperty('all'))) {
      delete ansibleGroupObj.all;
    }

    return ansibleGroupObj;
  }

  componentDidMount() {
    const {
      data: { mode, ansiblePlaybook, ansibleDataUrl, services, ansibleVarsAll, hostgroups },
      initApplicationDefinition,
      addApplicationDefinitionService,
      deleteApplicationDefinitionService,
      activateEditApplicationDefinitionService,
      changeEditApplicationDefinitionService,
      openForemanParameterSelectionModal,
      openAnsibleParameterSelectionModal,
      loadAnsibleData,
    } = this.props;

    const inlineEditButtonsFormatter = inlineEditFormatterFactory({
      isEditing: additionalData => this.props.editMode,
      renderValue: (value, additionalData) => (
        <td style={{ padding: '2px' }}>
          <Button
            bsStyle="default"
            onClick={() => activateEditApplicationDefinitionService(additionalData)}
          >
            <Icon type="pf" name="edit" title="edit entry" />
          </Button>
          &nbsp;
          <Button
            bsStyle="default"
            onClick={() => openForemanParameterSelectionModal(additionalData)}
          >
            <Icon type="pf" name="settings" title="change parameters" />
          </Button>
          &nbsp;
          <Button
            bsStyle="default"
            onClick={() => openAnsibleParameterSelectionModal(additionalData)}
          >
            <span title="change ansible variables">A</span>
          </Button>
          <DeleteTableEntry
            hidden={false}
            disabled={false}
            onDeleteTableEntry={deleteApplicationDefinitionService}
            additionalData={additionalData}
          />
        </td>
      ),
      renderEdit: (value, additionalData) => (
        <td style={{ padding: '2px' }}>
          <Button bsStyle="default" disabled>
            <Icon type="pf" name="edit" />
          </Button>
          &nbsp;
          <Button bsStyle="default" disabled>
            <Icon type="pf" name="settings" />
          </Button>
          &nbsp;
          <Button
            bsStyle="default" disabled>
            <span>A</span>
          </Button>
          <DeleteTableEntry
            hidden={false}
            disabled={true}
            onDeleteTableEntry={deleteApplicationDefinitionService}
            additionalData={additionalData}
          />
        </td>
      )
    });
    this.inlineEditButtonsFormatter = inlineEditButtonsFormatter;

    this.headerFormatter = EasyHeaderFormatter;

    const inlineEditFormatterImpl = {
      renderValue: (value, additionalData) => (
        <td>
          <span className="static">{value}</span>
        </td>
      ),
      renderEditText: (value, additionalData, subtype='text') => (
        <td className="editing">
          <FormControl
            type={subtype}
            defaultValue={value}
            onBlur={e => changeEditApplicationDefinitionService(e.target.value, additionalData) }
          />
        </td>
      ),
      renderEditSelect: (value, additionalData, options) => (
        <td className="editing">
          <Select
            value={value.toString()}
            onChange={e => changeEditApplicationDefinitionService(e.target.value, additionalData) }
            options={options}
            allowClear
            key="key"
          />
        </td>
      )
    };

    const inlineEditFormatter = inlineEditFormatterFactory({
      isEditing: additionalData => this.isEditing(additionalData),
      renderValue: (value, additionalData) => {
        let prettyValue = value;
        if (additionalData.property == 'hostgroup') {
          prettyValue = hostgroups[value];
        }
        else if (additionalData.property == 'ansibleGroup') {
          const ag = this.createAnsibleGroupObject(this.props.ansiblePlaybook.groups);
          prettyValue = ag[value];
        }
        return inlineEditFormatterImpl.renderValue(prettyValue, additionalData)
      },
      renderEdit: (value, additionalData) => {
        if (additionalData.property == 'hostgroup') {
          if (additionalData.rowData.newEntry === true) {
            return inlineEditFormatterImpl.renderEditSelect(value, additionalData, hostgroups);
          }
          return inlineEditFormatterImpl.renderValue(hostgroups[value], additionalData)
        }
        else if (additionalData.property == 'ansibleGroup') {
          const ag = this.createAnsibleGroupObject(this.props.ansiblePlaybook.groups);

          if (additionalData.rowData.newEntry === true) {
            return inlineEditFormatterImpl.renderEditSelect(value, additionalData, ag);
          }
          return inlineEditFormatterImpl.renderValue(ag[value], additionalData);
        }
        return inlineEditFormatterImpl.renderEditText(value, additionalData);
      }
    });
    this.inlineEditFormatter = inlineEditFormatter;

    initApplicationDefinition(
      ansiblePlaybook,
      services,
      ansibleVarsAll,
      this.headerFormatter,
      this.inlineEditFormatter,
      this.inlineEditButtonsFormatter,
    );
  };

  render() {
    const {
      data: { organization, location, mode, ansiblePlaybooks, foremanDataUrl, ansibleDataUrl },
      ansiblePlaybook,
      services,
      columns,
      addApplicationDefinitionService,
      confirmEditApplicationDefinitionService,
      cancelEditApplicationDefinitionService,
      closeForemanParameterSelectionModal,
      openAnsibleParameterSelectionModal,
      closeAnsibleParameterSelectionModal,
      ParameterSelectionModal,
      loadAnsibleData,
    } = this.props;

    return (
      <span>
        <div>
          <AnsiblePlaybookSelector
            label="Ansible Playbook"
            editable={ mode == 'newDefinition' }
            viewText={ ansiblePlaybook.name }
            options={ ansiblePlaybooks }
            onChange={ loadAnsibleData }
            selectValue={ ansiblePlaybook.id.toString() }
            additionalData={{url: ansibleDataUrl }}
          />
        </div>
        <div className="form-group">
          <AddTableEntry
            hidden={ false }
            disabled={ this.props.editMode }
            onAddTableEntry={ addApplicationDefinitionService }
          />
          <Table.PfProvider
            striped
            bordered
            hover
            dataTable
            inlineEdit
            columns={columns}
            components={{
              body: {
                row: Table.InlineEditRow,
                cell: cellProps => cellProps.children
              }
            }}
          >
            <Table.Header headerRows={resolve.headerRows({ columns })} />
            <Table.Body
              rows={services}
              rowKey="id"
              onRow={(rowData, { rowIndex }) => ({
                role: 'row',
                isEditing: () => this.isEditing({ rowData }),
                onCancel: () => cancelEditApplicationDefinitionService({ rowData, rowIndex }),
                onConfirm: () => confirmEditApplicationDefinitionService({ rowData, rowIndex }),
                last: rowIndex === services.length - 1
              })}
            />
          </Table.PfProvider>
          <AddTableEntry
            hidden={ false }
            disabled={ this.props.editMode }
            onAddTableEntry={ addApplicationDefinitionService }
          />
          <span style={{ marginLeft: 30 }}>
            Ansible group vars 'all':
            <Button
              style={{ marginLeft: 10 }}
              bsStyle="default"
              disabled={ this.props.editMode }
              onClick={() => openAnsibleParameterSelectionModal({
                isAllGroup: true
              })}
            >
              <span title="change ansible variables for 'all'">A</span>
            </Button>
          </span>
        </div>
        <div>
          <ForemanModal
            id="AppDefinitionForemanParamSelection"
            dialogClassName="param_selection_modal"
            title="Foreman Parameter definition for Application Definition"
          >
            <ForemanModal.Header closeButton={false}>
              Parameter definition
            </ForemanModal.Header>
            {this.props.parametersData ? (
              <ParameterSelection
                paramType={ PARAMETER_SELECTION_PARAM_TYPE_FOREMAN }
                location={ location }
                organization={ organization }
                paramDataUrl= { foremanDataUrl }
                data={ this.props.parametersData }
              />
            ) : (<span>Empty</span>)
            }
            <ForemanModal.Footer>
              <div>
                <Button bsStyle="primary" onClick={() => closeForemanParameterSelectionModal({ mode: 'save' })}>Save</Button>
                <Button bsStyle="default" onClick={() => closeForemanParameterSelectionModal({ mode: 'cancel' })}>Cancel</Button>
              </div>
            </ForemanModal.Footer>
          </ForemanModal>
        </div>
        <div>
          <ForemanModal
            id="AppDefinitionAnsibleParamSelection"
            dialogClassName="param_selection_modal"
            title="Ansible variables for Application Definition"
          >
            <ForemanModal.Header closeButton={false}>
              Parameter definition
            </ForemanModal.Header>
            {this.props.parametersData ? (
              <ParameterSelection
                paramType={ PARAMETER_SELECTION_PARAM_TYPE_ANSIBLE }
                location={ location }
                organization={ organization }
                data={ this.props.parametersData }
              />
            ) : (<span>Empty</span>)
            }
            <ForemanModal.Footer>
              <div>
                <Button bsStyle="primary" onClick={() => closeAnsibleParameterSelectionModal({ mode: 'save' })}>Save</Button>
                <Button bsStyle="default" onClick={() => closeAnsibleParameterSelectionModal({ mode: 'cancel' })}>Cancel</Button>
              </div>
            </ForemanModal.Footer>
          </ForemanModal>
        </div>
        <RailsData
          key='applications_definition'
          view='app_definition'
          parameter='services'
          value={JSON.stringify(this.props.services)}
        />
        <RailsData
          key='applications_definition'
          view='app_definition'
          parameter='ansible_vars_all'
          value={JSON.stringify(this.props.ansibleVarsAll)}
        />
      </span>
    )};
}

ApplicationDefinition.defaultProps = {
  error: {},
  editMode: false,
  ansiblePlaybook: { "id": '', "name": '' },
  services: [],
  ansibleVarsAll: [],
  parametersData: {},
  columns: [],
  editParamsOfRowId: null,
}

ApplicationDefinition.propTypes = {
  initApplicationDefinition: PropTypes.func,
  editMode: PropTypes.bool.isRequired,
  ansiblePlaybook: PropTypes.object,
  services: PropTypes.array,
  ansibleVarsAll: PropTypes.array,
  columns: PropTypes.array,
  addApplicationDefinitionService: PropTypes.func,
  deleteApplicationDefinitionService: PropTypes.func,
  activateEditApplicationDefinitionService: PropTypes.func,
  confirmEditApplicationDefinitionService: PropTypes.func,
  cancelEditApplicationDefinitionService: PropTypes.func,
  changeEditApplicationDefinitionService: PropTypes.func,
  openForemanParameterSelectionModal: PropTypes.func,
  closeForemanParameterSelectionModal: PropTypes.func,
  openAnsibleParameterSelectionModal: PropTypes.func,
  closeAnsibleParameterSelectionModal: PropTypes.func,
  parametersData: PropTypes.object,
};

export default ApplicationDefinition;
