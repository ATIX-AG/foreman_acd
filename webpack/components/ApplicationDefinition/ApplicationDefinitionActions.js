import React from 'react';
import api from 'foremanReact/API';

import {
  setModalOpen,
  setModalClosed,
} from 'foremanReact/components/ForemanModal/ForemanModalActions';

import {
  actionHeaderCellFormatter,
} from 'patternfly-react';

import {
  propsToSnakeCase,
  propsToCamelCase,
} from 'foremanReact/common/helpers';

import {
  APPLICATION_DEFINITION_INIT,
  APPLICATION_DEFINITION_CLOSE_ALERT_MODAL,
  APPLICATION_DEFINITION_LOAD_ANSIBLE_DATA_REQUEST,
  APPLICATION_DEFINITION_LOAD_ANSIBLE_DATA_SUCCESS,
  APPLICATION_DEFINITION_LOAD_ANSIBLE_DATA_FAILURE,
  APPLICATION_DEFINITION_SERVICE_DELETE,
  APPLICATION_DEFINITION_SERVICE_ADD,
  APPLICATION_DEFINITION_SERVICE_EDIT_ACTIVATE,
  APPLICATION_DEFINITION_SERVICE_EDIT_CONFIRM,
  APPLICATION_DEFINITION_SERVICE_EDIT_CHANGE,
  APPLICATION_DEFINITION_SERVICE_EDIT_CANCEL,
  APPLICATION_DEFINITION_FOREMAN_PARAMETER_SELECTION_MODAL_OPEN,
  APPLICATION_DEFINITION_FOREMAN_PARAMETER_SELECTION_MODAL_CLOSE,
  APPLICATION_DEFINITION_ANSIBLE_PARAMETER_SELECTION_MODAL_OPEN,
  APPLICATION_DEFINITION_ANSIBLE_PARAMETER_SELECTION_MODAL_CLOSE,
  APPLICATION_DEFINITION_CHANGE_PARAMETER_SELECTION_MODE,
} from './ApplicationDefinitionConstants';

import {
  supportedPluginsToHiddenParameterTypes,
} from '../../helper';

import {
  transformAnsiblePlaybook,
} from './ApplicationDefinitionHelper';

export const initApplicationDefinition = (
  ansiblePlaybook,
  services,
  ansibleVarsAll,
  supportedPlugins,
  headerFormatter,
  inlineEditFormatter,
  inlineEditButtonsFormatter,
) => dispatch => {
  const initialState = {};

  initialState.columns = [
    {
      property: 'name',
      header: {
        label: 'Name',
        formatters: [headerFormatter],
        props: {
          index: 0,
          style: {
            width: '15%'
          }
        },
      },
      cell: {
        formatters: [inlineEditFormatter]
      }
    },
    {
      property: 'description',
      header: {
        label: 'Description',
        formatters: [headerFormatter],
        props: {
          index: 1,
          style: {
            width: '10%'
          }
        },
      },
      cell: {
        formatters: [inlineEditFormatter]
      }
    },
    {
      property: 'hostgroup',
      header: {
        label: 'Hostgroup',
        formatters: [headerFormatter],
        props: {
          index: 2,
          style: {
            width: '20%'
          }
        },
      },
      cell: {
        formatters: [inlineEditFormatter]
      }
    },
    {
      property: 'ansibleGroup',
      header: {
        label: 'Ansible Group',
        formatters: [headerFormatter],
        props: {
          index: 3,
          style: {
            width: '20%'
          }
        },
      },
      cell: {
        formatters: [inlineEditFormatter]
      }
    },
    {
      property: 'minCount',
      header: {
        label: 'min count',
        formatters: [headerFormatter],
        props: {
          index: 4,
          style: {
            width: '10%'
          }
        },
      },
      cell: {
        formatters: [inlineEditFormatter]
      }
    },
    {
      property: 'maxCount',
      header: {
        label: 'max count',
        formatters: [headerFormatter],
        props: {
          index: 5,
          style: {
            width: '10%'
          }
        },
      },
      cell: {
        formatters: [inlineEditFormatter]
      }
    },
    {
      property: 'actions',
      header: {
        label: 'Actions',
        formatters: [actionHeaderCellFormatter],
        props: {
          index: 6,
          style: {
            width: '15%'
          }
        },
      },
      cell: {
        formatters: [inlineEditButtonsFormatter]
      }
    }
  ];

  if (ansiblePlaybook !== undefined) {
    initialState.ansiblePlaybook = transformAnsiblePlaybook(ansiblePlaybook);
  }
  initialState.services = services;
  initialState.ansibleVarsAll = ansibleVarsAll;
  initialState.hiddenForemanParameterTypes = supportedPluginsToHiddenParameterTypes(supportedPlugins);

  dispatch({
    type: APPLICATION_DEFINITION_INIT,
    payload: initialState,
  });
};

const errorHandler = (msg, err) => {
  const error = {
    errorMsg: 'Failed to fetch data from server.',
    statusText: err,
  };
  return { type: msg, payload: { error } };
};

export const closeAlertModal = () => ({
  type: APPLICATION_DEFINITION_CLOSE_ALERT_MODAL,
  payload: {}
});

export const loadAnsibleData = (
  ansiblePlaybookId,
  additionalData
) => dispatch => {
  dispatch({ type: APPLICATION_DEFINITION_LOAD_ANSIBLE_DATA_REQUEST });

  const baseUrl = additionalData.url;
  const realUrl = baseUrl.replace("__id__", ansiblePlaybookId);

  return api
    .get(realUrl, {}, {})
    .then(({ data }) =>
      dispatch({
        type: APPLICATION_DEFINITION_LOAD_ANSIBLE_DATA_SUCCESS,
        payload: { ...data }
      })
    )
    .catch(error => dispatch(errorHandler(APPLICATION_DEFINITION_LOAD_ANSIBLE_DATA_FAILURE, error)));
};

export const addApplicationDefinitionService = (additionalData) => ({
  type: APPLICATION_DEFINITION_SERVICE_ADD,
  payload: {
    ...additionalData,
  },
});

export const deleteApplicationDefinitionService = (additionalData) => ({
  type: APPLICATION_DEFINITION_SERVICE_DELETE,
  payload: {
    ...additionalData,
  },
});

export const activateEditApplicationDefinitionService = (additionalData) => ({
  type: APPLICATION_DEFINITION_SERVICE_EDIT_ACTIVATE,
  payload: {
    ...additionalData,
  },
});

export const confirmEditApplicationDefinitionService = (rowData) => ({
  type: APPLICATION_DEFINITION_SERVICE_EDIT_CONFIRM,
  payload: {
    ...rowData,
  },
});

export const cancelEditApplicationDefinitionService = (rowData) => ({
  type: APPLICATION_DEFINITION_SERVICE_EDIT_CANCEL,
  payload: {
    ...rowData,
  },
});

export const changeEditApplicationDefinitionService = (value, additionalData) => ({
  type: APPLICATION_DEFINITION_SERVICE_EDIT_CHANGE,
  payload: {
    value,
    ...additionalData,
  },
});

export const openForemanParameterSelectionModal = (additionalData) => dispatch => {
  dispatch({
    type: APPLICATION_DEFINITION_FOREMAN_PARAMETER_SELECTION_MODAL_OPEN,
    payload: {
      ...additionalData,
    }
  });
  dispatch(
    setModalOpen({ id: 'AppDefinitionForemanParamSelection' })
  );
}

export const closeForemanParameterSelectionModal = (additionalData) => dispatch => {
  dispatch({
    type: APPLICATION_DEFINITION_FOREMAN_PARAMETER_SELECTION_MODAL_CLOSE,
    payload: {
      ...additionalData,
    }
  });

  dispatch(
    setModalClosed({ id: 'AppDefinitionForemanParamSelection' })
  );
}

export const openAnsibleParameterSelectionModal = (additionalData) => dispatch => {
  dispatch({
    type: APPLICATION_DEFINITION_ANSIBLE_PARAMETER_SELECTION_MODAL_OPEN,
    payload: {
      ...additionalData,
    }
  });
  dispatch(
    setModalOpen({ id: 'AppDefinitionAnsibleParamSelection' })
  );
}

export const closeAnsibleParameterSelectionModal = (additionalData) => dispatch => {
  dispatch({
    type: APPLICATION_DEFINITION_ANSIBLE_PARAMETER_SELECTION_MODAL_CLOSE,
    payload: {
      ...additionalData,
    }
  });

  dispatch(
    setModalClosed({ id: 'AppDefinitionAnsibleParamSelection' })
  );
}

export const changeParameterSelectionMode = (additionalData) => ({
  type: APPLICATION_DEFINITION_CHANGE_PARAMETER_SELECTION_MODE,
  payload: {
    ...additionalData,
  },
})
