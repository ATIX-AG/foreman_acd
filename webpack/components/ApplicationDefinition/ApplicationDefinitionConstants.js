export const APPLICATION_DEFINITION_INIT = 'INIT_APPLICATION_DEFINITION_INIT';
export const APPLICATION_DEFINITION_SERVICE_DELETE = 'APPLICATION_DEFINITION_SERVICE_DELETE';
export const APPLICATION_DEFINITION_SERVICE_ADD = 'APPLICATION_DEFINITION_SERVICE_ADD';
export const APPLICATION_DEFINITION_SERVICE_EDIT_ACTIVATE = 'APPLICATION_DEFINITION_SERVICE_EDIT_ACTIVATE';
export const APPLICATION_DEFINITION_SERVICE_EDIT_CONFIRM = 'APPLICATION_DEFINITION_SERVICE_EDIT_CONFIRM';
export const APPLICATION_DEFINITION_SERVICE_EDIT_CHANGE = 'APPLICATION_DEFINITION_SERVICE_EDIT_CHANGE';
export const APPLICATION_DEFINITION_SERVICE_EDIT_CANCEL = 'APPLICATION_DEFINITION_SERVICE_EDIT_CANCEL';
export const APPLICATION_DEFINITION_PARAMETER_SELECTION_MODAL_OPEN = 'APPLICATION_DEFINITION_PARAMETER_SELECTION_MODAL_OPEN';
export const APPLICATION_DEFINITION_PARAMETER_SELECTION_MODAL_CLOSE = 'APPLICATION_DEFINITION_PARAMETER_SELECTION_MODAL_CLOSE';
