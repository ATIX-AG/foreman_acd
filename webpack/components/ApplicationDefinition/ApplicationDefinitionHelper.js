export function transformAnsiblePlaybook(playbook) {
  const ansiblePlaybook = new Object();
  ansiblePlaybook.id = playbook.id;
  ansiblePlaybook.name = playbook.name;
  ansiblePlaybook.groups = {};

  if (playbook.hasOwnProperty('groups')) {
    Object.entries(playbook.groups).forEach(([group_name, group_vars]) => {
      ansiblePlaybook.groups[group_name] = [];

      let id = 0;
      Object.entries(group_vars).forEach(([var_name, var_value]) => {
        const entry = {
          id,
          name: var_name,
          value: var_value,
        };
        ansiblePlaybook.groups[group_name].push(entry);
        id += 1;
      });
    });
  }

  return ansiblePlaybook;
}
