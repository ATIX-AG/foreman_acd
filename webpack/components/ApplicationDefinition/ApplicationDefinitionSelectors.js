const applicationDefinitionConf = state => state.foremanAcd.applicationDefinitionConf;

export const selectEditMode = state => applicationDefinitionConf(state).editMode;
export const selectAnsiblePlaybook = state => applicationDefinitionConf(state).ansiblePlaybook;
export const selectServices = state => applicationDefinitionConf(state).services;
export const selectColumns = state => applicationDefinitionConf(state).columns;
export const selectParametersData = state => applicationDefinitionConf(state).parametersData;
export const selectAnsibleVarsAll = state => applicationDefinitionConf(state).ansibleVarsAll;
export const selectParamEditMode = state => applicationDefinitionConf(state).paramEditMode;
