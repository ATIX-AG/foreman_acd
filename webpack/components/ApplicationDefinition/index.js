import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import './ApplicationDefinition.scss';
import ApplicationDefinition from './ApplicationDefinition';
import * as ApplicationDefinitionActions from './ApplicationDefinitionActions';

import {
  selectEditMode,
  selectAnsiblePlaybook,
  selectServices,
  selectColumns,
  selectParametersData,
  selectAnsibleVarsAll,
} from './ApplicationDefinitionSelectors';

const mapStateToProps = state => ({
  editMode: selectEditMode(state),
  ansiblePlaybook: selectAnsiblePlaybook(state),
  services: selectServices(state),
  columns: selectColumns(state),
  parametersData: selectParametersData(state),
  ansibleVarsAll: selectAnsibleVarsAll(state),
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(ApplicationDefinitionActions, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ApplicationDefinition);

