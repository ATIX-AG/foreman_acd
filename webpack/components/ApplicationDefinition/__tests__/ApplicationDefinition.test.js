import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import ApplicationDefinition from '../ApplicationDefinition';

const noop = () => {};

const fixtures = {
  'should render application definition': {
    data: {
      location: "Default Location",
      organization: "Default Organization",
      hostgroups: [],
      foremanDataUrl: "url/does/not/exist",
      ansibleDataUrl: "url/does/not/exist",
      services: [],
      ansibleVarsAll: [],
    },
    closeAlertModal: noop,
    loadAnsibleData: noop,
    initApplicationDefinition: noop,
    addApplicationDefinitionService: noop,
  },
};

describe('ApplicationDefinition', () =>
  testComponentSnapshotsWithFixtures(ApplicationDefinition, fixtures));
