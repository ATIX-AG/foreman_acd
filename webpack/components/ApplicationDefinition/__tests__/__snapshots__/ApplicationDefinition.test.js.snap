// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationDefinition should render application definition 1`] = `
<span>
  <MessageDialog
    accessibleDescription=""
    accessibleName=""
    className=""
    enforceFocus={true}
    footer={null}
    icon={
      <Icon
        name="error-circle-o"
        type="pf"
      />
    }
    onHide={[Function]}
    primaryAction={[Function]}
    primaryActionButtonBsStyle="danger"
    primaryActionButtonContent="OK"
    primaryContent=""
    secondaryAction={[Function]}
    secondaryActionButtonBsStyle="default"
    secondaryActionButtonContent={null}
    secondaryContent={null}
    show={false}
    title=""
  />
  <div>
    <AnsiblePlaybookSelector
      additionalData={
        Object {
          "url": "url/does/not/exist",
        }
      }
      editable={false}
      hidden={false}
      label="Ansible Playbook"
      onChange={[Function]}
      selectValue=""
      viewText=""
    />
    <div
      style={
        Object {
          "paddingTop": 25,
        }
      }
    >
      <pre>
        Ansible Playbook can't be blank
      </pre>
    </div>
  </div>
  <div
    className="form-group"
  >
    <TablePfProvider
      bordered={true}
      className=""
      columns={Array []}
      components={
        Object {
          "body": Object {
            "cell": [Function],
            "row": [Function],
          },
        }
      }
      condensed={false}
      dataTable={true}
      hover={true}
      inlineEdit={true}
      striped={true}
    >
      <Header
        headerRows={
          Array [
            Array [],
          ]
        }
      />
      <Body
        onRow={[Function]}
        rowKey="id"
        rows={Array []}
      />
    </TablePfProvider>
    <AddTableEntry
      disabled={true}
      hidden={false}
      onAddTableEntry={[Function]}
    />
    <span
      style={
        Object {
          "marginLeft": 30,
        }
      }
    >
      Ansible group vars 'all':
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        <span
          title="change ansible variables for 'all'"
        >
          A
        </span>
      </Button>
    </span>
  </div>
  <div>
    <ForemanModal
      dialogClassName="param_selection_modal"
      id="AppDefinitionForemanParamSelection"
      title="Foreman Parameter definition for Application Definition"
    >
      <Header
        closeButton={false}
      >
        Parameter definition
      </Header>
      <Connect(ParameterSelection)
        data={Object {}}
        editModeCallback={[Function]}
        location="Default Location"
        organization="Default Organization"
        paramDataUrl="url/does/not/exist"
        paramType="PARAMETER_SELECTION_PARAM_TYPE_FOREMAN"
      />
      <Footer>
        <div>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="primary"
            disabled={false}
            onClick={[Function]}
          >
            Save
          </Button>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            onClick={[Function]}
          >
            Cancel
          </Button>
        </div>
      </Footer>
    </ForemanModal>
  </div>
  <div>
    <ForemanModal
      dialogClassName="param_selection_modal"
      id="AppDefinitionAnsibleParamSelection"
      title="Ansible variables for Application Definition"
    >
      <Header
        closeButton={false}
      >
        Parameter definition
      </Header>
      <Connect(ParameterSelection)
        data={Object {}}
        editModeCallback={[Function]}
        location="Default Location"
        organization="Default Organization"
        paramType="PARAMETER_SELECTION_PARAM_TYPE_ANSIBLE"
      />
      <Footer>
        <div>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="primary"
            disabled={false}
            onClick={[Function]}
          >
            Save
          </Button>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            onClick={[Function]}
          >
            Cancel
          </Button>
        </div>
      </Footer>
    </ForemanModal>
  </div>
  <RailsData
    key="application_definition_services_data"
    parameter="services"
    value="[]"
    view="app_definition"
  />
  <RailsData
    key="application_definition_ansible_data"
    parameter="ansible_vars_all"
    value="[]"
    view="app_definition"
  />
</span>
`;
